### ISSUE TITLE
hh_client --remove-dead-fixmes removes ALL fixmes

### ISSUE THREAD

**Issue by ssandler** [facebook/hhvm#9043]
**Describe the bug**
Instead of only removing dead fixmes (which don't produce a type error when removed) this command now removes all of them.

**Standalone code, or other way to reproduce the problem**
```
$ git clone https://github.com/hhvm/hhast.git
$ cd hhast
$ composer install
$ docker run -it -v $(pwd):/hhast -w /hhast hhvm/hhvm:4.153-latest bash
$ hh_client --no-load
No errors!
$ hh_client --remove-dead-fixmes
$ hh_client
...
744 errors found.
```

**Expected behavior**

Only fixmes which can be removed without introducing typechecker errors are removed.

**Actual behavior**

Seemingly all fixmes are removed, introducing typechecker errors.

**Environment**
 - Operating system
Ubuntu 18.04
 - Installation method
hhvm/hhvm on dockerhub
 - HHVM Version
 ```
HipHop VM 4.153.0 (rel) (non-lowptr)
Compiler: 1647458518_157060578
Repo schema: 534e885b23eb8788358662b280cabd6799199f99
```

**Comment by Wilfred** [comment]
Yeah, it's removing too many fixmes in some cases and too few in others. Tracked as T125330792, we're seeing the same thing.

**Comment by 585855ghcch** [comment]
Ghzbbzh

**Comment by 585855ghcch** [comment]
Vsvv

**Comment by 585855ghcch** [comment]
git clone https://github.com/hhvm/hhast.git
$ cd hhast
$ composer install
$ docker run -it -v $(pwd):/hhast -w /hhast hhvm/hhvm:4.153-latest bash
$ hh_client --no-load
No errors!
$ hh_client --remove-dead-fixmes
$ hh_client
...
744 errors found.

**Comment by 585855ghcch** [comment]
@ssandler

**Comment by 585855ghcch** [comment]
@585855ghcch

**Comment by Samyra312007** [comment]
Respected Maintainers, If this issue is open then i would like to work on this issue.
